### DISCUSSION TITLE
Tag rule causing issues (unable to list the page tree, comment permissions ignored)

### DISCUSSION THREAD

**Discussion by nuraeil** [requarks/wiki#3787] [Error / Bug Report]
Hello everyone,

I am currently setting up permissions and I am facing an issue of having to seperate internal vs external content. Creating a Regex just seems too bothersome to maintain as the page structure will expand, meaning me having to maintain a Regex that might get extremly bloated. We do not need unnecessary overhead in maintaining additional filters for what is public and what is not. And as Wiki.js goes with "deny all unless you allow it", the Regex would need a lot of filters (20+). I know that it would be something simple like 

> \/(?:Path1|Path2|Path3|Path4|Pathforthe5thorso|Andadditionalpathforthe6)

and so on. But if the page titles get lengthy and additionally have a lot of those, then maintaining it will become a nightmare.

Edit: I tried using a Regex, which should only allow content that is at the root directory or in direct folders on the root-level to be allowed but it causes the site to redirect to the login.  The reason for this Regex is, that some of the names are reused again in internal folders and using a filter like above would cause the internal articles to be made public.

> ^(?![a-zA-Z])\/(Netzwerk|Medientechnik|Benutzerkonten|home).*

I also can not find any documentation on what Regex is allowed... 

That is why I wanted to use tags as a means but it does not seem to work at all. I tagged the pages that are meant to be public and setup the guest group that the pages tagged with public get read:pages, read:assets and read:comments.

The site simply throws 

> An unexpected error occured.

There is nothing written in the logs so... I guess this simply boils down to the Tag-Rule not working? I did not find anything in the official documentation regarding Tag-Rules. It does not even list it in the weighting of the rules:

![image](https://user-images.githubusercontent.com/32387849/113049608-1ea92e00-91a4-11eb-828b-238e2710e55d.png)

Using the approach of setting the rules with allow starting with / and then denying the internal things, would result in having to maintain the groups manually. There is a general group for everyong logging in through our Active Directory which allows writing comments as we do not want guests to comment. This group would then overwrite the allow permissions that everyone on our department would get, which would mean them not being able to edit anything in their sub-departments.

So... my question is: what is the intended usage of the Tag-Rule? Or is it even working? I am currently not sure if I should submit this as an issue or if I am simply trying to implement it the wrong way. 

If it is a bug, then I would submit an issue and hope that it gets fixed as we intend to start using Wiki.js next month and this would delay it a bit as the permissions will not be correct and I would have to find a workaround for it.

If it is not, would someone please be so kind as to explain to me the intended usage or otherwise suggest another way to setup the permissions for this?

Looking forward to your answers and thanks in advance!

Best regards,

Nuraeil

**Comment by nuraeil** [comment]
I had some time today to look into things... 

Here is what the dev console says on a page that has the tag public:
[dev-console-wikijs.log](https://github.com/Requarks/wiki/files/6273591/dev-console-wikijs.log)

And it mostly seems concerns comment section... Which is set in the rules. It also mentions the page tree...which seems odd...

`theme0.js?1617399708:2 Could not find current page in page tree listing!`

![image](https://user-images.githubusercontent.com/32387849/113912933-0062b400-97dc-11eb-8f07-c704dab2d006.png)

If I define a path-rule, it works and I am able to view the page as a guest.

The logs itself still show normal queries with no errors.

I also do not have anything configured in regards to Commento, Disqus or the default.

I will move this discussion to the issues.
_________________________________________
Host Info:

OS: Docker Container (Linux)
Wiki.js version: 2.5.201
Database engine: postgres 11.11 (postgresql-repmgr)
Node: 14.16.0
_________________________________________

Disabling comments does nothing in terms of fixing the issue... It still can not list the page tree, and there is a rotating bar in the upper right corner.

![image](https://user-images.githubusercontent.com/32387849/114142663-846b8780-9913-11eb-9b0a-8489282e689e.png)
![image](https://user-images.githubusercontent.com/32387849/114142682-89c8d200-9913-11eb-9d34-5e4b18769f24.png)
![image](https://user-images.githubusercontent.com/32387849/114142693-8d5c5900-9913-11eb-9fd3-18cec9aedade.png)

It will also not list the pages allowed with the tag rule to be displayed in the page tree.

This is the landing page (/de/home) which has 3 folders with pages that should be visible due to the pages having the tag public.

![image](https://user-images.githubusercontent.com/32387849/114142849-b2e96280-9913-11eb-9ce6-c669a598db38.png)

And this is one of the pages...

![image](https://user-images.githubusercontent.com/32387849/114142993-df04e380-9913-11eb-9956-2ffd80f531ea.png)

And these are the current rules for Guests:
![image](https://user-images.githubusercontent.com/32387849/114143439-72d6af80-9914-11eb-979a-f1628c8a3b47.png)


I can provide more information if it is needed.

**Comment by minionflo** [comment]
Im having the same problem
